### PR TITLE
feat: Report number of pending anchors in metrics and nodeStatus output

### PR DESCRIPTION
**Pull request by stbrody**

### CODE REVIEW

**Comment by stbrody** [comment]
FYI @3benbox

**Comment by 3benbox** [comment]
> FYI @3benbox

This looks great, but to be able to track it over time, we'd need a metric.

Can you wrap the `numPendingAnchorSubscriptions` such that is could be observed with the observability package?
https://github.com/ceramicnetwork/observability/blob/master/packages/service-metrics/src/service-metrics.ts#L253

File: packages/core/src/state-management/repository-internals.ts [hunk 1]
```diff
@@ -288,6 +299,8 @@ export class RepositoryInternals {
     anchorStatus$: Observable<CASResponse>
   ): Subscription {
     const stopSignal = new Subject<void>()
+    this.#numPendingAnchorSubscriptions++
+    Metrics.observe(ANCHOR_POLL_COUNT, this.#numPendingAnchorSubscriptions)
```
Comment (stbrody) [inline packages/core/src/state-management/repository-internals.ts:303]: there's no examples of `Metrics.observe` anywhere else in the codebase, so please let me know if I did this right @gvelez17

**Review by stbrody — COMMENTED** [review]
adding Golda to review Metrics change

**Review by ukstv — APPROVED** [review]

File: packages/common/src/utils/test-utils.ts [hunk 2]
```diff
@@ -84,7 +85,17 @@ export class TestUtils {
       now = new Date()
       await TestUtils.delay(100) // poll every 100ms
     }
-    onFailure(stream.state)
+    if (onFailure) {
+      onFailure(stream.state)
+    } else {
+      throw new Error(
```
Comment (gvelez17) [inline packages/common/src/utils/test-utils.ts:91]: great really glad to have this

**Review by gvelez17 — APPROVED** [review]
LGTM, nice to have this exposed

**Pull request merged**